apps/st2-history/controller.js

```diff
@@ -253,6 +256,46 @@ angular.module('main')
       }
     };
 
+    $scope.rerun = (function () {
+      var rerun = $scope.$new();
+
+      rerun.metaSpec = {
+        type: 'object',
+        properties: {
+          ref: {
+            type: 'string',
+            name: 'Action'
+          }
+        }
+      };
+
+
+      rerun.open = function () {
+        $scope.$root.state.go('^.rerun', {id: $scope.record.id});
+        rerun.payload = _.clone($scope.payload);
```

Comment:
re-run inherits all the parameters of the existing execution and if you want to override them you need to explicitly specify them - that's a feature.

If you want to use a default value, you need to explicitly override it like any other parameter. I'm definitely against using some kind of "magical" string to indicate a default value (e.g. null vs undefined, just no).

We do have a list runner types and list actions API so if you want to re-use a default value you need to call this first to retrieve a default value. I think that's perfectly fine and it's more explicit. And so far I haven't had a use case when I would want to re-run and action and use a default parameter value.